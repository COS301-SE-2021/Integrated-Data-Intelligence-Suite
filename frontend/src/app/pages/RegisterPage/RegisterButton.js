import {Button} from 'antd';
import React, {Component} from 'react';
// import {validateLoginDetails} from "./functions/validateLoginDetails";

class RegisterButton extends React.Component {
    constructor(props) {
        super(props);
    }

    state = {
        is_loading: false,
        username: '',
        password: '',
    }

    enterLoading(val) {

        this.setState({
            is_loading: !this.state.is_loading,
        })

        this.setState({
            is_loading: false,
        })

        // console.log(user)
    }

    render() {
        return (
            <>
                <Button
                    type="primary"
                    htmlType="submit"
                    className="login_button"
                    loading={this.state.is_loading}
                    onClick={() => this.enterLoading(!this.state.is_loading)}
                >
                    Register
                </Button>
            </>
        );
    }
}

export default RegisterButton;