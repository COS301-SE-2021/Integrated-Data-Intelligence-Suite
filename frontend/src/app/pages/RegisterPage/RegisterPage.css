.login_button {
    background: #5873f9;
    border-color: #5773FA;
    width: 100%;
}

.forgot_password_link {
    color: var(--neon-blue);
    float: none;
}

/*The  Lines in between the word OR*/
div.ant-divider.ant-divider-horizontal.ant-divider-with-text.ant-divider-with-text-center.or_divider{
    display: flex;
    margin: 16px 0;
    color: black !important;
    font-weight: 500;
    font-size: 16px;
    white-space: nowrap;
    text-align: center;
    border-top: 0;
    border-top-color: black!important;
}

.register_link {
    color: var(--neon-blue);
    float: right;
}

.back-to-register{
    color: var(--neon-blue);
    float: left;
}

/*!* The root div. Its encompasses all the elements/components *!*/
/*#root {*/
/*    !*border: 3px solid red;*!*/
/*    border: 0;*/
/*    height: 100vh;*/
/*    position: relative;*/
/*    margin:auto;*/
/*    !*width: 100vw;*!*/
/*    !*background: white;*!*/
/*}*/

.forgot_password_link_container > .ant-form-item-control > .ant-form-item-control-input
> .ant-form-item-control-input-content{
    text-align: center;
}

#register_card{
    /*border: 3px solid #5773FA;*/
    /* align-self: center; */
    border:none;

    /*Aligning it Horizontally in the Center*/
    width: 30%;
    min-width: 389px;

    /* Aligning it Vertically in the center  */
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%)
}

#register-background {
    /*border: 3px solid red;*/
    height: 100vh;
    width: 100vw;
    display: grid;
    grid-template-columns: 100%;
    grid-template-rows: 100%;
    padding: 10%;
    justify-content: center;
    align-items: center;
    background: #27818e;
    z-index: 2;

}

#register-form-container {
    /*border: 3px solid blue;*/
    display: grid;
    grid-template-rows: 100%;
    grid-template-columns: 50% 50%;
    background-color: white;
    z-index: 2;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

#register-card {
    grid-column-start: 1;
    grid-column-end: 2;
    display: grid;
    grid-template-columns: 100%;
    grid-template-rows: auto auto;
    background-color: white;
    padding: 18%;
}

#register-card-title {
    grid-row-start: 1;
    grid-row-end: 2;
    font-weight: 600;
    font-size: 18px;
    color: #4169E1;
    margin-bottom: 5%;
}

#register-form {
    grid-row-start: 2;
    grid-row-end: 3;
}

#register-card-svg-bg {
    grid-column-start: 2;
    grid-column-end: 3;
    background: url("../../assets/BackgroundPatterns/Whangaehu.svg");
    background-color: white;
    background-size: cover;
}

.background.gradient.register{
    height: 100vh;
    padding: 0;
    margin: 0;
    /*background-image: linear-gradient(176deg, #1c8efa 70%, #000000 calc(70% + 2px));*/
}

/***********************************************/
/***************Custom Background***************/
/***********************************************/

#login-custom-bg {
    z-index: 1;
    width: 100vw;
    height: 100vh;
    position: absolute;
    background: white;
    /*border: 3px solid red;*/
    display: grid;
    grid-template-columns: 50% 50%;
    grid-template-rows: 50% 50%;
}

#top-left-block {
    grid-column-start: 1;
    grid-column-end: 2;
    grid-row-start: 1;
    grid-row-end: 2;
    background: rgb(1,49,152);
    background: linear-gradient(90deg, rgba(1,49,152,1) 0%, rgba(29,148,214,1) 100%);
}

#top-right-block {
    grid-column-start: 2;
    grid-column-end: 3;
    grid-row-start: 1;
    grid-row-end: 2;
    background: rgb(29,148,214);
    background: linear-gradient(90deg, rgba(29,148,214,1) 0%, rgba(140,223,255,1) 100%);

}

#bottom-right-block {
    grid-column-start: 2;
    grid-column-end: 2;
    grid-row-start: 2;
    grid-row-end: 2;
    background: rgb(29,148,214);
    background: linear-gradient(90deg, rgba(29,148,214,1) 0%, rgba(140,223,255,1) 100%);
}

#bottom-left-block {
    grid-column-start: 1;
    grid-column-end: 2;
    grid-row-start: 2;
    grid-row-end: 2;
    background: rgb(1,49,152);
    background: linear-gradient(90deg, rgba(1,49,152,1) 0%, rgba(29,148,214,1) 100%);
}


