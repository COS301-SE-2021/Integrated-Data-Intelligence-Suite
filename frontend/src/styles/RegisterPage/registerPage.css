#register_card {
    border: 3px solid deeppink;
    /* align-self: center; */

    /*Aligning it Horizontally in the Center*/
    width: 50%;
    /*max-width: 300px;*/

    /* Aligning it Vertically in the center  */
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%)
}

.login_button {
    background: deeppink;
    border-color: deeppink;
    width: 100%;
}

.forgot_password_link {
    color: deeppink;
    float: none;
}

/*The  Lines in between the word OR*/
.ant-divider-horizontal.ant-divider-with-text {
    display: flex;
    margin: 16px 0;
    color: black;
    font-weight: 500;
    font-size: 16px;
    white-space: nowrap;
    text-align: center;
    border-top: 0;
    border-top-color: rgb(0 0 0);
}

.register_link {
    color: deeppink;
    float: right;
}

/* The root div. Its encompasses all the elements/components */
#root {
    border: 3px solid red;
    height: 100vh;
    position: relative;
    margin:auto;
    /*width: 100vw;*/
    background: white;
}

.forgot_password_link_container > .ant-form-item-control > .ant-form-item-control-input
> .ant-form-item-control-input-content{
    text-align: center;
}


