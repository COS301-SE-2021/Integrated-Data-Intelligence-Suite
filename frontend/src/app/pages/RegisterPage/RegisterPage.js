import {useFormik} from 'formik';
import React, {Component} from 'react';
import RegisterButton from "../../components/RegisterButton/RegisterButton";
import {Link, useHistory} from "react-router-dom";
import {UserOutlined, LockOutlined} from '@ant-design/icons';
import {Form, Input, Button, Checkbox, Card, Divider} from 'antd';
import './registerPage.css';


//Validation Function
const validate = (values) => {
    const errors = {};

    //email validation
    if (!values.email) {
        errors.email = 'Required';
    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
        errors.email = 'Invalid email address';
    }

    //firstname validation
    if (!values.firstName) {
        errors.firstName = 'Required';
    }

    //lastname validation
    if (!values.lastName) {
        errors.lastName = 'Required';

    }

    //lastname validation
    if (!values.username) {
        errors.username = 'Required';
    }

    //password validation
    if (!values.password) {
        errors.password = 'required'
    }

    //confirmed password validation
    if (values.confirmedpassword !== values.password) {
        errors.confirmedpassword = 'passwords don\'t match';
    }
    return errors;
}


const RegisterPage = () => {
    localStorage.clear();
    let history = useHistory();
    const formik = useFormik({
        initialValues: {
            email: '',
            lastName: '',
            username: '',
            password: '',
            firstName: ''
        },
        validate,
        onSubmit: values => {
            //alert(JSON.stringify(values, null, 2));
            const requestOptions = {
                method: 'POST',
                headers: {'Content-Type': 'application/json'},
                body: JSON.stringify(values)
            };
            fetch('/user/register', requestOptions)
                .then(response => {
                    return response.json()
                }).then(json => {
                    if(json.success) {
                        history.push('/');
                    }else{
                        alert(json.message)
                    }
                });

            //use this to go to another page after successful validation server-side

        },

    });

    return (
        <div className={"background gradient register"}>
        <Card
            id={"register_card"}
            className={"loginCard"}
            title="Register"
        >
            <form onSubmit={formik.handleSubmit}>

                <Form.Item
                    label={'First Name'}
                >
                    <Input
                        id="firstName"
                        name="firstName"
                        type="text"
                        placeholder="first name"
                        onChange={formik.handleChange}
                        onBlur={formik.handleBlur}
                        value={formik.values.firstName}
                        // prefix={<UserOutlined className="site-form-item-icon"/>}
                    />
                    {formik.touched.firstName && formik.errors.firstName ? (
                        <p>{formik.errors.firstName}</p>) : null}
                </Form.Item>

                <Form.Item
                    label={'Last Name'}
                >
                    <Input
                        id="lastName"
                        name="lastName"
                        type="text"
                        placeholder="last name"
                        onChange={formik.handleChange}
                        onBlur={formik.handleBlur}
                        value={formik.values.lastName}
                        // prefix={<UserOutlined className="site-form-item-icon"/>}
                    />
                    {formik.touched.lastName && formik.errors.lastName ? (
                        <p>{formik.errors.lastName}</p>) : null}
                </Form.Item>

                <Form.Item
                    label={'User name'}
                >
                    <Input
                        id="username"
                        name="username"
                        type="text"
                        placeholder="user name"
                        onChange={formik.handleChange}
                        onBlur={formik.handleBlur}
                        value={formik.values.username}
                        // prefix={<UserOutlined className="site-form-item-icon"/>}
                    />
                    {formik.touched.username && formik.errors.username ? (
                        <p>{formik.errors.username}</p>) : null}
                </Form.Item>

                <Form.Item
                    label={'email'}
                >
                    <Input
                        id="email"
                        name="email"
                        type="email"
                        placeholder="email"
                        onChange={formik.handleChange}
                        onBlur={formik.handleBlur}
                        value={formik.values.email}
                        // prefix={<UserOutlined className="site-form-item-icon"/>}
                    />
                    {formik.touched.email && formik.errors.email ? (
                        <p>{formik.errors.email}</p>) : null}
                </Form.Item>

                <Form.Item
                    label={'password'}
                >
                    <Input.Password
                        id="password"
                        name="password"
                        type="password"
                        placeholder="Password"
                        value={formik.values.password}
                        onChange={formik.handleChange}
                        onBlur={formik.handleBlur} //When the user leaves the form
                        // prefix={<LockOutlined className="site-form-item-icon"/>}
                    />
                    {formik.touched.password && formik.errors.password ? (
                        <p>{formik.errors.password}</p>) : null}

                </Form.Item>

                <Form.Item
                    label={'Confirm password'}
                >
                    <Input.Password
                        id="confirmedpassword"
                        name="confirmedpassword"
                        type="password"
                        placeholder="rewrite the password"
                        value={formik.values.confirmedpassword}
                        onChange={formik.handleChange}
                        onBlur={formik.handleBlur} //When the user leaves the form
                        // prefix={<LockOutlined className="site-form-item-icon"/>}
                    />
                    {formik.touched.confirmedpassword && formik.errors.confirmedpassword ? (
                        <p>{formik.errors.confirmedpassword}</p>) : null}

                </Form.Item>

                <Form.Item>
                    <RegisterButton/>
                </Form.Item>

                <Divider className={'or_divider'}>
                    OR
                </Divider>

                <Form.Item>
                    Already have an account?
                    <Link to={"/login"}>
                        <a className={"register_link"} href="#">login</a>
                    </Link>
                </Form.Item>

                {/*<Form.Item*/}
                {/*    className={"forgot_password_link_container"}*/}
                {/*>*/}
                {/*    <a className="forgot_password_link" href="">*/}
                {/*        Forgot passw*/}
                {/*    </a>*/}
                {/*</Form.Item>*/}
            </form>
        </Card>
        </div>
    );
};


export default RegisterPage;
