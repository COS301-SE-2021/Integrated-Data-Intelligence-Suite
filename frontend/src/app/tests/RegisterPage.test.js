import React from 'react';
import renderer from 'react-test-renderer';
import '../Mocks/matchMedia.mock';
import { BrowserRouter } from 'react-router-dom';
import { unmountComponentAtNode } from 'react-dom';
import RegisterPage from '../pages/RegisterPage/RegisterPage';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <RegisterPage />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
