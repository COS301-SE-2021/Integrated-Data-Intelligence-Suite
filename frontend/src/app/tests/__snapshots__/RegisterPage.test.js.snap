// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="background gradient register"
>
  <div
    className="ant-card ant-card-bordered loginCard"
    id="register_card"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Register
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title="First Name"
            >
              First Name
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <input
                  className="ant-input"
                  id="firstName"
                  name="firstName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="first name"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title="Last Name"
            >
              Last Name
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <input
                  className="ant-input"
                  id="lastName"
                  name="lastName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="last name"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title="User name"
            >
              User name
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <input
                  className="ant-input"
                  id="username"
                  name="username"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="user name"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title="email"
            >
              email
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <input
                  className="ant-input"
                  id="email"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="email"
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title="password"
            >
              password
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <span
                  className="ant-input-affix-wrapper ant-input-password"
                  onMouseUp={[Function]}
                >
                  <input
                    action="click"
                    className="ant-input"
                    id="password"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Password"
                    style={null}
                    type="password"
                    value=""
                  />
                  <span
                    className="ant-input-suffix"
                  >
                    <span
                      aria-label="eye-invisible"
                      className="anticon anticon-eye-invisible ant-input-password-icon"
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      role="img"
                      tabIndex={-1}
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="eye-invisible"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                        />
                        <path
                          d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title="Confirm password"
            >
              Confirm password
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <span
                  className="ant-input-affix-wrapper ant-input-password"
                  onMouseUp={[Function]}
                >
                  <input
                    action="click"
                    className="ant-input"
                    id="confirmedpassword"
                    name="confirmedpassword"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="rewrite the password"
                    style={null}
                    type="password"
                    value=""
                  />
                  <span
                    className="ant-input-suffix"
                  >
                    <span
                      aria-label="eye-invisible"
                      className="anticon anticon-eye-invisible ant-input-password-icon"
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      role="img"
                      tabIndex={-1}
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="eye-invisible"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                        />
                        <path
                          d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <button
                  className="ant-btn ant-btn-primary login_button"
                  onClick={[Function]}
                  type="submit"
                >
                  <span>
                    Register
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center or_divider"
          role="separator"
        >
          <span
            className="ant-divider-inner-text"
          >
            OR
          </span>
        </div>
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                Already have an account?
                <a
                  href="/login"
                  onClick={[Function]}
                >
                  <a
                    className="register_link"
                    href="#"
                  >
                    login
                  </a>
                </a>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
